### PR TITLE
#1491 - Introduction of ExecReload option for (some) SystemD services

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

File: ovs/dal/migration/ovsmigrator.py [hunk 1]
```diff
@@ -339,4 +339,27 @@ def migrate(previous_version):
                                 junction_partition.partition.roles.append(DiskPartition.ROLES.DTL)
                                 junction_partition.partition.save()
 
+            # Version 16: Addition of 'ExecReload' for AlbaProxy SystemD services
+            getattr(ServiceManager, 'has_service')  # Invoke ServiceManager to fill out the ImplementationClass (default None)
```
Comment (khenderick) [inline ovs/dal/migration/ovsmigrator.py]: Should be moved to some async migration logic. Let's start this effort with some scheduled task in some kind of migrationcontroller that runs every day or so.

**Review by khenderick — CHANGES_REQUESTED** [review]

File: ovs/lib/migration.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                root_client.run(['systemctl', 'daemon-reload'])
+
+        # Adjustment of open file descriptors for Arakoon services to 8192
+        changed_clients = set()
+        for storagerouter in StorageRouterList.get_storagerouters():
+            root_client = sr_client_map[storagerouter.guid]
+            for service_name in ServiceManager.list_services(client=root_client):
+                if not service_name.startswith('ovs-arakoon-'):
+                    continue
+                if '8192' in root_client.file_read(filename='/lib/systemd/system/{0}.service'.format(service_name)):
```
Comment (kvanhijf) [inline ovs/lib/migration.py]: This won't work for systems using Upstart

**Review by kvanhijf — COMMENTED** [review]
See comment

**Review by kvanhijf — COMMENTED** [review]
Approved

**Review by khenderick — APPROVED** [review]

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1492 --> openvstorage-2.7.13-dev.1488538734.dd3d2df
https://github.com/openvstorage/alba-asdmanager/pull/227 --> openvstorage-sdm-1.6.13-dev.1488538802.6a23c62

**Pull request merged**